opentera-webrtc-native-client/OpenteraWebrtcNativeClient/include/OpenteraWebrtcNativeClient/Utils/FunctionTask.h

```diff
@@ -36,34 +22,26 @@ namespace opentera
             }
             else
             {
-                FunctionTask task(function);
-                thread->PostTask(std::unique_ptr<QueuedTask>(&task));
-                task.m_event.Wait(rtc::Event::kForever);
-                return std::move(task.m_returnedValue);
+                rtc::Event event;
+                T returnedValue;
+                thread->PostTask(
+                    RTC_FROM_HERE,
+                    [&]()
+                    {
+                        returnedValue = std::move(function());
+                        event.Set();
+                    });
+                event.Wait(rtc::Event::kForever);
+                return std::move(returnedValue);
```

Comment:
`std::move` can be detrimental here, it can prevent copy elision
https://en.cppreference.com/w/cpp/language/return#Automatic_move_from_local_variables_and_parameters